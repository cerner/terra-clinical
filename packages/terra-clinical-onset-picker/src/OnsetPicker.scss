:local {
  .fieldset {
    margin-bottom: 0;

    > div {
      margin-bottom: 0;
      margin-top: 0;
    }
  }

  .field-inline {
    margin-bottom: 0.357em;
    margin-right: 0.357em;
  }
}
