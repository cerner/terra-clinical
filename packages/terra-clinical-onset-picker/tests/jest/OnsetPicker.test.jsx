import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import MockDate from 'mockdate';
import { IntlProvider } from 'react-intl';
import selectMessages from 'terra-form-select/translations/en-US.json';

import OnsetPicker from '../../lib/OnsetPicker';
import onsetMessages from '../../translations/en-US.json';

const locale = 'en-US';
const messages = Object.assign({}, selectMessages, onsetMessages);

beforeEach(() => {
  MockDate.set('11/15/2017');
});

afterEach(() => {
  MockDate.reset();
});

it('should render a default onset picker with specified onset date', () => {
  const onsetPicker = (
    <IntlProvider locale={locale} messages={messages}>
      <OnsetPicker
        birthdate="2011-08-16T17:40:49-05:00"
        id="test"
        onsetDate="2017-09-10T17:40:49-05:00"
      />
    </IntlProvider>
  );
  expect(render(onsetPicker)).toMatchSnapshot();
});

it('should render only a select when supplied with the UNKNOWN precision', () => {
  const onsetPicker = (
    <IntlProvider locale={locale} messages={messages}>
      <OnsetPicker
        birthdate="2011-08-16T17:40:49-05:00"
        id="test"
        precision="unknown"
      />
    </IntlProvider>);
  expect(render(onsetPicker)).toMatchSnapshot();
});

it('should render with non default inputs when supplied', () => {
  const onsetPicker = (
    <IntlProvider locale={locale} messages={messages}>
      <OnsetPicker
        birthdate="2011-08-16T17:40:49-05:00"
        granularity="year"
        id="test"
        precision="before"
        onsetDate="2014-08-16T17:40:49-05:00"
      />
    </IntlProvider>);
  expect(render(onsetPicker)).toMatchSnapshot();
});

it('should render the same when onChange functions are supplied', () => {
  const onsetPicker = (
    <IntlProvider locale={locale} messages={messages}>
      <OnsetPicker
        birthdate="2011-08-16T17:40:49-05:00"
        granularitySelectOnChange={() => {}}
        id="test"
        precisionSelectOnChange={() => {}}
        onsetDateInputOnChange={() => {}}
      />
    </IntlProvider>);
  expect(render(onsetPicker)).toMatchSnapshot();
});

it('should render only the supplied precisions', () => {
  const onsetPicker = (
    <IntlProvider locale={locale} messages={messages}>
      <OnsetPicker
        birthdate="2011-08-16T17:40:49-05:00"
        id="test"
        precisionSet={['unknown', 'before', 'after']}
      />
    </IntlProvider>);
  expect(render(onsetPicker)).toMatchSnapshot();
});

it('throws error on missing locale prop in Base', () => {
  const onsetPicker = (
    <OnsetPicker
      birthdate="2011-08-16T17:40:49-05:00"
      id="test"
    />
  );

  try {
    shallow(onsetPicker);
  } catch (e) {
    expect(e.message).toContain('add locale prop to Base component');
  }
});
