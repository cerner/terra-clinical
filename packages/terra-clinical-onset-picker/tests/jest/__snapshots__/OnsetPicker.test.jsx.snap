// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default onset picker with specified onset date 1`] = `
<div
  id="test"
>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline precision"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-precision-select"
            id="test-precision-select-label"
          >
            Select a date precision
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-4"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-1 terra-select-screen-reader-display-2 terra-select-screen-reader-placeholder-3"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-precision-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-1"
            >
              Select a date precision
            </span>
            <span
              id="terra-select-screen-reader-description-4"
            >
              List of 5 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <div
              class="placeholder"
              id="terra-select-screen-reader-placeholder-3"
            >
              Precision
            </div>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
      <div
        class="field field-inline granularity"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-granularity-select"
            id="test-granularity-select-label"
          >
            Select a date granularity
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-8"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-5 terra-select-screen-reader-display-6 terra-select-screen-reader-placeholder-7"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-granularity-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-5"
            >
              Select a date granularity
            </span>
            <span
              id="terra-select-screen-reader-description-8"
            >
              List of 4 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <div
              class="placeholder"
              id="terra-select-screen-reader-placeholder-7"
            >
              Granularity
            </div>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    />
  </fieldset>
</div>
`;

exports[`should render only a select when supplied with the UNKNOWN precision 1`] = `
<div
  id="test"
>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline precision"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-precision-select"
            id="test-precision-select-label"
          >
            Select a date precision
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-12"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-9 terra-select-screen-reader-display-10 terra-select-screen-reader-placeholder-11"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-precision-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-9"
            >
              Select a date precision
            </span>
            <span
              id="terra-select-screen-reader-description-12"
            >
              List of 5 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <span
              id="terra-select-screen-reader-display-10"
            >
              Unknown
            </span>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`should render only the supplied precisions 1`] = `
<div
  id="test"
>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline precision"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-precision-select"
            id="test-precision-select-label"
          >
            Select a date precision
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-48"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-45 terra-select-screen-reader-display-46 terra-select-screen-reader-placeholder-47"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-precision-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-45"
            >
              Select a date precision
            </span>
            <span
              id="terra-select-screen-reader-description-48"
            >
              List of 3 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <div
              class="placeholder"
              id="terra-select-screen-reader-placeholder-47"
            >
              Precision
            </div>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
      <div
        class="field field-inline granularity"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-granularity-select"
            id="test-granularity-select-label"
          >
            Select a date granularity
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-52"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-49 terra-select-screen-reader-display-50 terra-select-screen-reader-placeholder-51"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-granularity-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-49"
            >
              Select a date granularity
            </span>
            <span
              id="terra-select-screen-reader-description-52"
            >
              List of 4 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <div
              class="placeholder"
              id="terra-select-screen-reader-placeholder-51"
            >
              Granularity
            </div>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    />
  </fieldset>
</div>
`;

exports[`should render the same when onChange function is supplied 1`] = `
<div
  id="test"
>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline precision"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-precision-select"
            id="test-precision-select-label"
          >
            Select a date precision
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-40"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-37 terra-select-screen-reader-display-38 terra-select-screen-reader-placeholder-39"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-precision-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-37"
            >
              Select a date precision
            </span>
            <span
              id="terra-select-screen-reader-description-40"
            >
              List of 5 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <div
              class="placeholder"
              id="terra-select-screen-reader-placeholder-39"
            >
              Precision
            </div>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
      <div
        class="field field-inline granularity"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-granularity-select"
            id="test-granularity-select-label"
          >
            Select a date granularity
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-44"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-41 terra-select-screen-reader-display-42 terra-select-screen-reader-placeholder-43"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-granularity-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-41"
            >
              Select a date granularity
            </span>
            <span
              id="terra-select-screen-reader-description-44"
            >
              List of 4 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <div
              class="placeholder"
              id="terra-select-screen-reader-placeholder-43"
            >
              Granularity
            </div>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    />
  </fieldset>
</div>
`;

exports[`should render with age inputs filled in when supplied 1`] = `
<div
  id="test"
>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline precision"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-precision-select"
            id="test-precision-select-label"
          >
            Select a date precision
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-28"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-25 terra-select-screen-reader-display-26 terra-select-screen-reader-placeholder-27"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-precision-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-25"
            >
              Select a date precision
            </span>
            <span
              id="terra-select-screen-reader-description-28"
            >
              List of 5 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <span
              id="terra-select-screen-reader-display-26"
            >
              Before
            </span>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
      <div
        class="field field-inline granularity"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-granularity-select"
            id="test-granularity-select-label"
          >
            Select a date granularity
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-32"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-29 terra-select-screen-reader-display-30 terra-select-screen-reader-placeholder-31"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-granularity-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-29"
            >
              Select a date granularity
            </span>
            <span
              id="terra-select-screen-reader-description-32"
            >
              List of 4 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <span
              id="terra-select-screen-reader-display-30"
            >
              Age
            </span>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline age"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-age-input"
            id="test-age-input-label"
          >
            Enter an age
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <input
          aria-labelledby="test-age-input-label"
          class="form-input"
          id="test-age-input"
          max="6"
          min="0"
          step="1"
          type="number"
          value="3"
        />
      </div>
      <div
        class="field field-inline age-unit"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-age-unit-select"
            id="test-age-unit-select-label"
          >
            Select an age precision
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-36"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-33 terra-select-screen-reader-display-34 terra-select-screen-reader-placeholder-35"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-age-unit-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-33"
            >
              Select an age precision
            </span>
            <span
              id="terra-select-screen-reader-description-36"
            >
              List of 3 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <span
              id="terra-select-screen-reader-display-34"
            >
              Year(s)
            </span>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`should render with non default inputs when supplied 1`] = `
<div
  id="test"
>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline precision"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-precision-select"
            id="test-precision-select-label"
          >
            Select a date precision
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-16"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-13 terra-select-screen-reader-display-14 terra-select-screen-reader-placeholder-15"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-precision-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-13"
            >
              Select a date precision
            </span>
            <span
              id="terra-select-screen-reader-description-16"
            >
              List of 5 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <span
              id="terra-select-screen-reader-display-14"
            >
              Before
            </span>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
      <div
        class="field field-inline granularity"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-granularity-select"
            id="test-granularity-select-label"
          >
            Select a date granularity
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-20"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-17 terra-select-screen-reader-display-18 terra-select-screen-reader-placeholder-19"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-granularity-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-17"
            >
              Select a date granularity
            </span>
            <span
              id="terra-select-screen-reader-description-20"
            >
              List of 4 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <span
              id="terra-select-screen-reader-display-18"
            >
              Year
            </span>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="fieldset fieldset"
  >
    <div
      class="fieldset-children"
    >
      <div
        class="field field-inline year"
      >
        <div
          class="label-group label-group-hidden"
        >
          <label
            class="label"
            for="test-year-select"
            id="test-year-select-label"
          >
            Select a year
          </label>
          <div
            class="error-icon-hidden"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </div>
        </div>
        <div
          aria-describedby="terra-select-screen-reader-description-24"
          aria-disabled="false"
          aria-expanded="false"
          aria-haspopup="true"
          aria-labelledby="terra-select-screen-reader-label-21 terra-select-screen-reader-display-22 terra-select-screen-reader-placeholder-23"
          class="select default"
          data-terra-select="true"
          data-terra-select-combobox="true"
          id="test-year-select"
          role="combobox"
          tabindex="0"
        >
          <div
            class="visually-hidden-component"
            hidden=""
          >
            <span
              id="terra-select-screen-reader-label-21"
            >
              Select a year
            </span>
            <span
              id="terra-select-screen-reader-description-24"
            >
              List of 7 options. Use up and down arrow keys to navigate through options. Press enter to select an option.
            </span>
          </div>
          <div
            class="display"
            role="textbox"
          >
            <span
              id="terra-select-screen-reader-display-22"
            >
              2014
            </span>
          </div>
          <div
            class="toggle"
            data-terra-form-select-toggle="true"
          >
            <span
              class="arrow-icon"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            aria-relevant="additions text"
            class="visually-hidden-component"
          />
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;
